lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -74,6 +74,24 @@ struct ConstantOpConv : public OpConversionPattern<ConstantOp> {
   }
 };
 
+struct NamedConstantOpConv : public OpConversionPattern<NamedConstantOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(NamedConstantOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // make it as global if it is a parameter kind
+    if (op.getKind() == moore::NamedConst::Parameter) {
+      rewriter.setInsertionPointToStart(
+          op->getParentOfType<mlir::ModuleOp>()->getBlock());
```

Comment:
It's not legal to refer values in the global scope from IsolatedAbove op. I think you can just create the op in the local scope.

Reply:
Thanks a lot. I  just met a crash when iserting.